vm/src/obj/objlist.rs

```diff
@@ -153,6 +153,26 @@ pub fn list_extend(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     Ok(vm.get_none())
 }
 
+fn list_index(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    trace!("list.index called with: {:?}", args);
+    arg_check!(
+        vm,
+        args,
+        required = [(list, Some(vm.ctx.list_type())), (needle, None)]
+    );
+    for (index, element) in get_elements(list).iter().enumerate() {
+        match vm.call_method(needle, "__eq__", vec![element.clone()]) {
```

Comment:
Use here the `?` operator to propagate the error outwards. Like this: `let value = vm.call_method(needle, "__eq__", vec![element.clone()])?;`